lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs

```diff
@@ -85,6 +85,16 @@ protected ConfigurationBuilderImpl(IConfiguration[] confs) : this(new string[0],
         {
         }
 
+        public IConfigurationBuilder AddOptionalConfiguration(IConfiguration conf)
```

Comment:
lets answer the question if we want to support notion of optional configuration at all.
I felt it could be convenient.
As for allowing null implicitly - I don't think this is good approach as it may lead to ambiguous configurations. 
While with AddOptionalConfiguraiton method, client has explicit way to tell which configuration is optional and has less chances to create buggy configuration.